### PR TITLE
Fix configuration test

### PR DESCRIPTION
**Pull request by CarlesDD**
### What does this PR do?
Fix configuration test.

### Motivation
Some test for configuration were not using correct assertion. They were void and failing.

### Plugin Checklist

- [x] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### Additional Notes
The modified test used the assertion `to.deep.include` in order to check the first argument of the first call to `updateConfig`.

This assertion was actually not testing anything, since the expected value (all the config entries) should be wrapped in an array as a first argument.

### CODE REVIEW

File: packages/dd-trace/test/config.spec.js [hunk 1]
```diff
@@ -270,21 +269,32 @@ describe('Config', () => {
       { name: 'profiling.sourceMap', value: true, origin: 'default' },
       { name: 'profiling.exporters', value: 'agent', origin: 'default' },
       { name: 'startupLogs', value: false, origin: 'default' },
-      { name: 'telemetry.enabled', value: true, origin: 'default' },
+      { name: 'telemetry.enabled', value: true, origin: 'env_var' },
```
Comment (CarlesDD) [inline packages/dd-trace/test/config.spec.js:272]: Is `env_var` the correct value here?

File: packages/dd-trace/test/config.spec.js [hunk 2]
```diff
@@ -270,21 +269,32 @@ describe('Config', () => {
...
+        value: '(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?)key)|token|consumer_?(?:id|key|secret)|sign(?:ed|ature)|bearer|authorization',
+        origin: 'default'
+      },
+      {
+        name: 'appsec.obfuscatorValueRegex',
+        // eslint-disable-next-line max-len
+        value: '(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?|access_?|secret_?)key(?:_?id)?|token|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)(?:\\s*=[^;]|"\\s*:\\s*"[^"]+")|bearer\\s+[a-z0-9\\._\\-]+|token:[a-z0-9]{13}|gh[opsu]_[0-9a-zA-Z]{36}|ey[I-L][\\w=-]+\\.ey[I-L][\\w=-]+(?:\\.[\\w.+\\/=-]+)?|[\\-]{5}BEGIN[a-z\\s]+PRIVATE\\sKEY[\\-]{5}[^\\-]+[\\-]{5}END[a-z\\s]+PRIVATE\\sKEY|ssh-rsa\\s*[a-z0-9\\/\\.+]{100,}',
+        origin: 'default'
+      },
+      { name: 'remoteConfig.enabled', value: true, origin: 'env_var' },
```
Comment (CarlesDD) [inline packages/dd-trace/test/config.spec.js:297]: Is `env_var` the correct origin value for `remoteConfig.enabled`?

File: packages/dd-trace/test/config.spec.js [hunk 3]
```diff
@@ -293,12 +303,28 @@ describe('Config', () => {
       { name: 'iast.deduplicationEnabled', value: true, origin: 'default' },
       { name: 'iast.redactionEnabled', value: true, origin: 'default' },
       { name: 'iast.telemetryVerbosity', value: 'INFORMATION', origin: 'default' },
+      { name: 'iast.redactionNamePattern', value: null, origin: 'default' },
+      { name: 'iast.redactionValuePattern', value: null, origin: 'default' },
       { name: 'isCiVisibility', value: false, origin: 'default' },
       { name: 'gitMetadataEnabled', value: true, origin: 'default' },
       { name: 'openaiSpanCharLimit', value: 128, origin: 'default' },
-      { name: 'traceId128BitGenerationEnabled', value: false, origin: 'default' },
-      { name: 'traceId128BitLoggingEnabled', value: false, origin: 'default' }
```
Comment (CarlesDD) [inline packages/dd-trace/test/config.spec.js:300]: These values were duplicated.

File: packages/dd-trace/test/config.spec.js (hunk 2 above)
Comment (uurien) [reply packages/dd-trace/test/config.spec.js:297]: Same here,
```
    this._setBoolean(env, 'remoteConfig.enabled', coalesce(
      DD_REMOTE_CONFIGURATION_ENABLED,
      !this._isInServerlessEnvironment()
    ))
```

where:
```
!this._isInServerlessEnvironment()
```

Not in this PR, as here you're only enabling the tests, but I think that we should move the `this._isInServerlessEnvironment()` to `default` values and not to `env`.

**Review by uurien — COMMENTED** [review]
I made some comments but LGTM

I am not approving it because this need review also of the rest of the teams

File: packages/dd-trace/test/config.spec.js [hunk 4]
```diff
@@ -243,8 +244,8 @@ describe('Config', () => {
       { name: 'site', value: 'datadoghq.com', origin: 'default' },
       { name: 'hostname', value: '127.0.0.1', origin: 'default' },
       { name: 'port', value: '8126', origin: 'default' },
-      { name: 'debug', value: false, origin: 'default' },
```
Comment (simon-id) [inline packages/dd-trace/test/config.spec.js:246]: does debug not exists anymore ?

**Review by juan-fernandez — APPROVED** [review]
looks good from ci vis' perspective

File: packages/dd-trace/test/config.spec.js (hunk 4 above)
Comment (CarlesDD) [reply packages/dd-trace/test/config.spec.js:246]: It is not present in the first argument of the call to `updateConfig`.

File: packages/dd-trace/test/config.spec.js [hunk 5]
```diff
@@ -514,11 +540,12 @@ describe('Config', () => {
       { name: 'traceId128BitGenerationEnabled', value: true, origin: 'env_var' },
       { name: 'traceId128BitLoggingEnabled', value: true, origin: 'env_var' },
       { name: 'spanAttributeSchema', value: 'v1', origin: 'env_var' },
-      { name: 'spanComputePeerService', value: true, origin: 'env_var' },
       { name: 'sampler.rateLimit', value: '-1', origin: 'env_var' },
       { name: 'spanRemoveIntegrationFromService', value: true, origin: 'env_var' },
-      { name: 'peerServiceMapping', value: 'c:cc,d:dd', origin: 'env_var' },
-      { name: 'tracePropagationStyle.extract', value: ['b3', 'tracecontext'], origin: 'env_var' },
+      { name: 'peerServiceMapping', value: { c: 'cc', d: 'dd' }, origin: 'env_var' },
```
Comment (tlhunter) [inline packages/dd-trace/test/config.spec.js:545]: What happened here? Was the old test testing at a different level of abstraction, like before stuff was parsed?

Comment (CarlesDD) [reply packages/dd-trace/test/config.spec.js:545]: The old testing was not asserting any of these entries, so my guess is the assertion was always wrong.

**Review by Qard — APPROVED** [review]

**Comment by simon-id** [comment]
once this is merged i'll do another PR to sort the entries alphabetically like in config.js because any change like this is unreviewable without it

**Review by uurien — APPROVED** [review]

**Pull request merged**